### DISCUSSION TITLE
Framesync via semaphore posting from a "tick" thread

### DISCUSSION THREAD

**Discussion by theotherraycole** [DistroAV/DistroAV#965] [Ideas]
I have code that uses the existing framesync code but corrects the sleep call with a wait for a semaphore from a tick thread. Below is the source code. Search for "pulseSem" and you'll see the relevant code. I have no idea if this works with streams that have audio but I presume it does...but since I have not tested it and don't have a good NDI source to test against it (I use video-only NDI sources) I don't want to submit a pull request.

https://github.com/theotherraycole/obs-ndi/blob/643d1e63ccd2f4b7f7e7d4a7d87066fc7786a1d2/src/obs-ndi-source.cpp

**Comment by paulpv** [comment]
Thanks @theotherraycole
We were just talking about your code today on our Discord! :)
https://discord.gg/ZuTxbUK3ug

**Comment by Jaybonaut** [comment]
Forgive me for asking; what does the Framesync (Experimental) feature do exactly?  Is it worth turning on?
